superset-frontend/src/explore/components/DatasourcePanel.tsx

```diff
@@ -0,0 +1,197 @@
+  mapStateToProps: QueryFormData;
+  type: ControlType;
+  label: string;
+  datasource?: DatasourceControl;
+}
+
+interface Props {
+  datasource: {
+    columns: Array<any>;
+    metrics: Array<any>;
+  };
+  controls: {
+    datasource: DatasourceControl;
+  };
+  actions: Partial<ExploreActions> & Pick<ExploreActions, 'setControlValue'>;
+}
+
+const DatasourceContainer = styled.div`
+  background-color: ${({ theme }) => theme.colors.grayscale.light4};
+  position: relative;
+  height: 100%;
+  .field-selections {
+    padding: 0 ${({ theme }) => 2 * theme.gridUnit}px;
+  }
+  .ant-collapse
+    > .ant-collapse-item
+    > .ant-collapse-header
+    .ant-collapse-arrow {
+    right: ${({ theme }) => theme.gridUnit * -50}px;
+  }
+  .ant-collapse > .ant-collapse-item > .ant-collapse-header {
+    padding-left: 10px;
+    padding-bottom: 0px;
+  }
+  .form-control.input-sm {
+    margin-bottom: ${({ theme }) => theme.gridUnit * 3}px;
+  }
+  .ant-collapse-item {
+    background-color: ${({ theme }) => theme.colors.grayscale.light4};
+    .anticon.anticon-right.ant-collapse-arrow > svg {
+      transform: rotate(90deg) !important;
+    }
+  }
+  .ant-collapse-item.ant-collapse-item-active {
+    .anticon.anticon-right.ant-collapse-arrow > svg {
+      transform: rotate(-90deg) !important;
+    }
+  }
+  .header {
+    font-size: ${({ theme }) => theme.typography.sizes.l}px;
+    margin-left: ${({ theme }) => theme.gridUnit * -2}px;
+  }
+  .ant-collapse-content-box > div {
+    margin-left: -14px;
+  }
+  .type-label {
+    text-align: left;
+  }
+  .metric-option .option-label {
+    margin-left: ${({ theme }) => theme.gridUnit * -5}px;
+  }
+  .field-selections {
+    position: absolute;
+    top: ${({ theme }) => theme.gridUnit * 15}px;
+    bottom: 0;
+    left: 0;
+    right: 0;
+    overflow: auto;
+  }
+  .field-length {
+    margin-top: -3px;
+    margin-bottom: ${({ theme }) => theme.gridUnit * 2}px;
+    font-size: ${({ theme }) => theme.typography.sizes.s}px;
+    color: ${({ theme }) => theme.colors.grayscale.light1};
+  }
+`;
+
+const DataSourcePanel = ({
+  datasource,
+  controls: { datasource: datasourceControl },
+  actions,
+}: Props) => {
+  const { columns, metrics } = datasource;
+  const [lists, setList] = useState({
+    columns,
+    metrics,
+  });
+  const search = ({ target: { value } }: { target: { value: string } }) => {
+    if (value === '') {
+      setList({ columns, metrics });
+      return;
+    }
+    const filteredColumns = lists.columns.filter(
+      column => column.column_name.indexOf(value) !== -1,
+    );
+    const filteredMetrics = lists.metrics.filter(
+      metric => metric.metric_name.indexOf(value) !== -1,
+    );
+    setList({ columns: filteredColumns, metrics: filteredMetrics });
+  };
+  useEffect(() => {
+    setList({
+      columns,
+      metrics,
+    });
+  }, [datasource]);
+
+  const metricSlice = lists.metrics.slice(0, 50);
+  const columnSlice = lists.columns.slice(0, 50);
```

Comment:
is there anyway to see more than the first 50 of these? Pagination or something like that?

Reply:
As of now there is no pagination. We are using the search bar as a way for users to search through large columns and metrics.

Reply:
I think we can just use virtual list without pagination.

Reply:
I spent some time trying to get the virtual list working, but it requires a prop with the `height` value. Since this is all CSS-based layout (hooray!) this got difficult quickly, using refs and more lifecycle hooks than I'd care to fiddle with in this PR. 

I think I see a path forward with `ParentSize` from `@vx/responsive` but that might wing up being a fast-follow PR.

Reply:
With 50, we have feature parity... Hope that's OK for a minute, but we'll try to address it ASAP (like... today or tomorrow, ideally)

Reply:
Since we want to add resizable columns soon, you can probably just use fixed width based on `window.clientWidth` instead of `ParentSize`.